### PR TITLE
add postgres exporter

### PR DESCRIPTION
**Pull request by blupman**
#### Pull Request (PR) description
I have added a postgres_exporter module, the module is installing https://github.com/wrouesnel/postgres_exporter

### CODE REVIEW

File: manifests/postgres_exporter.pp [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+#  This can be file, or env (default 'env')
+#
+#  [*data_source_uri*]
+#  Uri on howto connect to the database
+
+class prometheus::postgres_exporter (
+  String $download_extension,
+  String $download_url_base,
+  Array[String] $extra_groups,
+  String $group,
```
Comment (bastelfreak) [inline manifests/postgres_exporter.pp]: can you update the `String` datatype  to `String[1]` on all locations where it makes sense? (It enforces a minimum string length of 1).

File: spec/classes/postgres_exporter_spec.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+            version: '0.4.6',
+            arch: 'amd64',
+            os: 'linux',
+            bin_dir: '/usr/local/bin',
+            install_method: 'url'
+          }
+        end
+
+        describe 'install correct binary' do
+          it { is_expected.to contain_file('/usr/local/bin/postgres_exporter').with('target' => '/opt/postgres_exporter-0.4.6.linux-amd64/postgres_exporter') }
```
Comment (bastelfreak) [inline spec/classes/postgres_exporter_spec.rb]: please also add something like this:
```
it { is_expected.to compile.with_all_deps }
```

**Comment by bastelfreak** [comment]
hey @blupman, thanks for the PR! I did some little inline comments, but all in all this looks good. Travis is currently having some issues, but I hope that will be fixed soon.

File: manifests/postgres_exporter.pp (hunk 1 above)
Comment (blupman) [reply manifests/postgres_exporter.pp]: I've added quite a few [1]'s to where it makes sense. also some minor cleanups, in de comments

File: spec/classes/postgres_exporter_spec.rb (hunk 2 above)
Comment (bastelfreak) [reply spec/classes/postgres_exporter_spec.rb]: That would be awesome!

File: data/defaults.yaml [hunk 3]
```diff
@@ -253,6 +253,18 @@ prometheus::blackbox_exporter::package_name: 'blackbox_exporter'
 prometheus::blackbox_exporter::modules: {}
 prometheus::blackbox_exporter::config_file: '/etc/blackbox-exporter.yaml'
 prometheus::blackbox_exporter::version: '0.7.0'
+prometheus::postgres_exporter::data_source_uri: 'host=/var/run/postgresql/ sslmode=disable'
+prometheus::postgres_exporter::postgres_pass: 'postgres_exporter_password'
+prometheus::postgres_exporter::postgres_user: 'postgres_exporter_username'
```
Comment (bastelfreak) [inline data/defaults.yaml]: should postgres_user and postgres_pass default to undef? Might makes it more obvious to users that they need to change it.

File: manifests/postgres_exporter.pp [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+#  method for presenting username and password to the exporter
+#  This can be file, or env (default 'env')
+#
+#  [*data_source_uri*]
+#  Uri on howto connect to the database
+
+class prometheus::postgres_exporter (
+  String[1] $download_extension,
+  String[1] $download_url_base,
+  Array[String] $extra_groups,
```
Comment (bastelfreak) [inline manifests/postgres_exporter.pp]: Can you please update this to `Array[String[1]]`? That will enforce a minimum string length of 1.

File: manifests/postgres_exporter.pp [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  String[1] $user,
+  String[1] $version,
+  String[1] $data_source_uri,
+  String[1] $postgres_pass,
+  String[1] $postgres_user,
+  Enum['env', 'file'] $postgres_auth_method,
+  Boolean $purge_config_dir      = true,
+  Boolean $restart_on_change     = true,
+  Boolean $service_enable        = true,
+  String $service_ensure         = 'running',
```
Comment (bastelfreak) [inline manifests/postgres_exporter.pp]: for all the other String datatypes here: Please ensure a minimal String length of 1.

**Comment by bastelfreak** [comment]
@blupman can you recheck again please? I made some more inline comments.

File: manifests/postgres_exporter.pp (hunk 5 above)
Comment (blupman) [reply manifests/postgres_exporter.pp]: now its done for all non optional strings.

File: manifests/postgres_exporter.pp (hunk 4 above)
Comment (blupman) [reply manifests/postgres_exporter.pp]: ok. done

File: data/defaults.yaml (hunk 3 above)
Comment (blupman) [reply data/defaults.yaml]: agreed, 
i had to add the postgres_pass and postgres_pass to the spec test, to make the daemon start

**Comment by bastelfreak** [comment]
@blupman can you take a look at the failing spec tests?

**Comment by blupman** [comment]
everyhing should be fine now... (i've finaly made spent some time on 
createing a local env for running spec tests)

**Comment by blupman** [comment]
i should work now, i found some time to make the non acceptance 
spectests work localy, now invest a bit of time in makine acceptance 
tests work locally

**Comment by bastelfreak** [comment]
Thanks!

File: manifests/postgres_exporter.pp [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+#  please refer to https://github.com/wrouesnel/postgres_exporter
+#
+#  [*group*]
+#  Group under which the binary is running
+#
+#  [*init_style*]
+#  Service startup scripts style (e.g. rc, upstart or systemd)
+#
+#  [*install_method*]
+#  Installation method: url or package (only url is supported currently)
```
Comment (alexjfisher) [inline manifests/postgres_exporter.pp:39]: `only url is supported currently`.  Is that true?

Comment (blupman) [reply manifests/postgres_exporter.pp:39]: I'm not aware of a packaged version of postgres_exporter, if it exsist, its not tested.

**Pull request merged**